src/lib/isSlug.js

```diff
@@ -0,0 +1,18 @@
+import assertString from './util/assertString';
+
+let charset = '^([a-z0-9\\-]{1,})$';
+let regex_charset = new RegExp(`${charset}`);
+let regex_boundaries_consecutive = /[-_]{2,}/;
+let regex_boundaries_leading = /^[-_]/;
```

Comment:
Good work on this. Here are my comments and some changes request: 

1. Use of camel casing e.g 
  ```javascript 
let regexCharset = new RegExp(`${charset}`)
```
2. We can make some changes instead of using `new Regex` example provided 
```javascript
let charsetRegex = /^([a-z0-9\\-]{1,})$/;
// remove let regex_charset = new RegExp(`${charset}`);
// Other code remains the same
return (
      charsetRegex.test(str) && 
      // Other code here
 )
```
3. Also, we can make the regex further combined
```javascript
import assertString from './util/assertString';

let charsetRegex = /^[^-_](?!.*?[-_]{2,})([a-z0-9\\-]{1,}).*[^-_]$/;

export default function isSlug(str) {
  assertString(str);
  return (charsetRegex.test(str));
}
```

Reply:
thanks for this Feedback @ezrqnkemboi, makes good use of the re-usable variable.